nixos/modules/services/backup/bacula.nix

```diff
@@ -108,13 +128,95 @@ let
     ${dir_cfg.extraConfig}
     '';
 
+  tlsOptions = {...}:
+  {
+    options = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Specifies if TLS should be enabled if this is false TLS will be
+          completely disabled, even if tls.require is true.
+        '';
+      };
+      require = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = mdDoc ''
+          Require TLS or TLS-PSK encryption. This directive is ignored unless
+          one of TLS Enable or TLS PSK Enable is set to true.
+          If TLS is not required while TLS or TLS-PSK are enabled, then the
+          Bacula component will connect with other components either with or
+          without TLS or TLS-PSK
+
+          If TLS or TLS-PSK is enabled and TLS is required, then the Bacula
+          component will refuse any connection request that does not use TLS.
+        '';
+      };
+      certificate = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = mdDoc ''
+          The full path and filename of a PEM encoded TLS certificate.
+          It will be used as either a client or server certificate,
+          depending on the connection direction.
+          This directive is required in a server context, but it may
+          not be specified in a client context if TLS Verify Peer is
+          set to `false` in the corresponding server context.
+        '';
+      };
+      key = mkOption {
+        type = types.str;
+        description = mdDoc ''
+          The full path and filename of a PEM encoded TLS
+          private key. It must correspond to the TLS certificate.
+        '';
+      };
```

Comment:
I think "path" describes "full path and filename" enough.
You can also probably use `types.path` instead of `types.str`.